sharedmem.c

```diff
@@ -34,7 +34,7 @@ static int parse_hex_sysattr(struct udev_device *dev, const char *name,
 		return -ENOENT;
 
 	errno = 0;
-	val = strtoul(buf, &endptr, 16);
+	val = strtoull(buf, &endptr, 16);
 	if ((val == LONG_MAX && errno == ERANGE) || endptr == buf) {
```

Comment:
Not sure why the original code has a LONG_MAX check here, but the LONG_MAX check may be invalid depending on the size of phys_addr_t.

Reply:
Oh, this looks like an attempt at handling this piece of the man page:

RETURN VALUE
       The strtoul() function returns either the result of the conversion  or,
       if  there  was  a leading minus sign, the negation of the result of the
       conversion represented as an unsigned value, unless the original  (non‐
       negated)  value  would  overflow; in the latter case, strtoul() returns
       ULONG_MAX and sets errno to ERANGE.  Precisely the same holds for  str‐
       toull() (with ULLONG_MAX instead of ULONG_MAX).


Except it should have been looking for ULONG_MAX (and now ULLONG_MAX). I'll fix that part.